server/etcdserver/v3_server.go

```diff
@@ -270,6 +270,16 @@ func (s *EtcdServer) LeaseGrant(ctx context.Context, r *pb.LeaseGrantRequest) (*
 	return resp.(*pb.LeaseGrantResponse), nil
 }
 
+func (s *EtcdServer) waitApplyPendingCommits() error {
```

Comment:
Mismatch between function name and what it does. Function does really wait for applied commits, however uses  `leasehttp.ApplyTimeout` which is specific for leases. We should either rename function to be lease specific, or have timeout as an argument.

Reply:
Just changed the method name to applyWaitWithTimeout, which has a timeout parameter. thx